### DISCUSSION TITLE
(adding to a feature in c# 10) Exclude assembly references for a certain file

### DISCUSSION THREAD

**Discussion by just-a-hriday** [dotnet/csharplang#5064] [Language Ideas]
# Context

As of .NET 6 preview 5, C# 10 has global namespaces - you can put a using statement in one place, and that namespace will be "used" for all files in the current project. Here's an example:
```c#
global using System;
global using System.Collections.Generic;
global using System.Linq;
```
# Suggestion

But what if you want to exclude a namespace just for one file?

## Motivation

This can be really useful if there are two global namespace imports which have classes with the same name. 

For example, let's say there's a namespace `Foo` and a namespace `Bar`. They both have classes named `Foobar` - so `Foo.Foobar` and `Bar.Foobar`. Foo and Bar are both globally imported. 
Now, let's say another class wants to use `Bar.Foobar` and never uses `Foo.Foobar`. But currently, as Foo and Bar are both globally imported, the class would have to say `Bar.Foobar` every time - that completely ruins the point of actually writing a `using` statement for the namespace at all!

For people bringing up the fact that you can just _not_ globally import both `Foo` and `Bar`, it's possible that most classes need both of them, and it's only our that example class needs just `Bar`.

---
So, I want to suggest
## New syntax

This should be possible:
```c#
!using Foo;
// or
exclude Foo;
// or something similar
```

**Comment by svick** [comment]
One workaround that works today it to use an alias:

```c#
using Foobar = Bar.Foobar;
```

Though this doesn't work well if it's many types with this problem, not just one.
